crates/nu-protocol/src/value/record.rs

```diff
@@ -98,15 +92,19 @@ impl Record {
     }
 
     pub fn get(&self, col: impl AsRef<str>) -> Option<&Value> {
-        self.index_of(col).and_then(|idx| self.vals.get(idx))
+        self.inner
+            .iter()
+            .find_map(|(k, v)| if k == col.as_ref() { Some(v) } else { None })
```

Comment:
I recently just learned about `bool::then_some`. This could be
```suggestion
            .find_map(|(k, v)| (k == col.as_ref()).then_some(v))
```
if you think that this is better.

Reply:
Can go either way, always wondering what the impact on the perf of the **debug** builds is with those fun little adapters.